ResetPassword.js

```diff
@@ -64,6 +150,33 @@ module.exports = function (Model, options) {
                     });
                 }
             });
+        });
+		/**
+		 * The following route handles password reset for phone.
+		 */
+        Model.app.post('/confirm-password-reset-phone', function(request, response, next) {
+            /*
+             * Find the token data corresponding to the given 
+             * token.
+             */
+            Model.relations.accessTokens.modelTo.findById(request.query.access_token, (err, token) => {
```

Comment:
I'm unclear who is responsible for making sure that `request.query.access_token` exists and if not then sending back a http error code like 422? I'm not saying your code is wrong, I'm just pointing out something I don't understand.

Reply:
The token comes through `confirm-password.html` when the value of query param `phone` is `true`. Here, we are checking whether the sent `access_token` is valid or not. If it doesn't exist then the request is valid and we return a `401` response. If the `access_token` is valid, we proceed with the desired operation.